src/ParticleDA.jl

```diff
@@ -516,87 +202,51 @@ function tdac(params::Parameters, rng::AbstractVector{<:Random.AbstractRNG})
     end
     timer = TimerOutput()
 
-    @timeit_debug timer "Initialization" begin
-
-        states, statistics, observations, stations, weights, field_buffer = init_arrays(params)
+    nprt_per_rank = Int(params.nprt / MPI.Comm_size(MPI.COMM_WORLD))
 
-        background_grf = init_gaussian_random_field_generator(params)
-
-        # Set up tsunami model
-        model_matrices = LLW2d.setup(params.nx,
-                                     params.ny,
-                                     params.bathymetry_setup,
-                                     params.absorber_thickness_fraction,
-                                     params.boundary_damping,
-                                     params.cutoff_depth)
+    # Do memory allocations
 
-        set_stations!(stations, params)
+    # TODO: ideally this will be an argument of the function, to choose a
+    # different datatype.
+    T = Float64
 
-        set_initial_state!(states, model_matrices, field_buffer, rng, nprt_per_rank, params)
+    if MPI.Comm_rank(MPI.COMM_WORLD) == params.master_rank
+        weights = Vector{T}(undef, params.nprt)
+    else
+        weights = Vector{T}(undef, nprt_per_rank)
+    end
 
-        resampling_indices = Vector{Int}(undef,params.nprt)
+    resampling_indices = Vector{Int}(undef, params.nprt)
 
-        avg_arr = Array{Float64,3}(undef, params.nx, params.ny, params.n_state_var)
-        var_arr = Array{Float64,3}(undef, params.nx, params.ny, params.n_state_var)
+    # TODO: these variables should be set in a better way
+    nx, ny, n_state_var = params.nx, params.ny, params.n_state_var
+    statistics = Array{SummaryStat{T}, 3}(undef, nx, ny, n_state_var)
+    avg_arr = Array{T,3}(undef, nx, ny, n_state_var)
+    var_arr = Array{T,3}(undef, nx, ny, n_state_var)
```

Comment:
I think the sizes of these arrays should come from the model.  It needs to somehow pass the size `(nx, ny, n_state_var)` to the filter.